pkg/controller/node/controller_utils.go

```diff
@@ -80,6 +83,30 @@ func deletePods(kubeClient clientset.Interface, recorder record.EventRecorder, n
 	return remaining, nil
 }
 
+// deleteSinglePod will delete the pod that is running on given node from master,
+// and return true if the pod was deleted, or was found pending deletion.
+func deleteSinglePod(kubeClient clientset.Interface, recorder record.EventRecorder, pod *api.Pod, daemonStore cache.StoreToDaemonSetLister) (bool, error) {
+	// if the pod has already been marked for deletion, we still return true that there are remaining pods.
```

Comment:
I think this comment is out-of-date, and referring to the previous version of this function where it was evicting nodes ("that there are remaining pods"). I think you can just remove the whole comment, since the function-level comment explains that you return true if the pod was found pending deletion.